src/pages/contact/ContactForm.tsx

```diff
@@ -0,0 +1,131 @@
+import styled from 'styled-components';
+import Button from '../../components/Button';
+import React, { useRef, useState } from 'react';
+import emailjs from '@emailjs/browser';
+
+interface ContactProps {
+    info: {
+        email: string;
+        social: {
+            icon: {
+                src: string;
+                alt: string;
+            };
+            link: string;
+        }[];
+    }
+}
+
+
+const ContactForm = (props: ContactProps) => {
+    const form = useRef<any>(null);
```

Comment:
using `any` is bad practice so to be more specific we should use HTMLFormElement and check if form.current is not null
```ts
    const form = useRef<HTMLFormElement>(null);
    const [showSent, setSent] = useState(false)

    const sendEmail = (e: React.FormEvent) => {
        e.preventDefault();
        if (form.current) {
            emailjs.sendForm('service_xgab7p1', 'template_hkk20pi', form.current, 'YT9q6oUsSjEmsZASM').then((result) => {
                console.log(result.text);
                form.current.reset();
                setSent(true);
            }, (error) => {
                console.log(error.text);
                //to display error text on page
            });
        }
    }
```

but here we will get a TS error for `form.current.reset();` because `form.current` may be null inside the sendEmail callback.

there are 2 ways to fix this:
1. use the optional chaining operator: `form.current?.reset()`
2. store `form.current` in a const after ensuring it's not null: 
```ts
    const form = useRef<HTMLFormElement>(null);
    const [showSent, setSent] = useState(false)

    const sendEmail = (e: React.FormEvent) => {
        e.preventDefault();
        if (form.current) {
            const current = form.current;
            emailjs.sendForm('service_xgab7p1', 'template_hkk20pi', current, 'YT9q6oUsSjEmsZASM').then((result) => {
                console.log(result.text);
                current.reset();
                setSent(true);
            }, (error) => {
                console.log(error.text);
                //to display error text on page
            });
        }
    }
```